### PR TITLE
vs create flavor test

### PR DESCRIPTION
**Pull request by FernandoOjeda**
Fixed vs create with flavor data test https://github.com/softlayer/softlayer-python/issues/1033

### CODE REVIEW

File: SoftLayer/CLI/virt/create.py [hunk 1]
```diff
@@ -334,6 +335,20 @@ def cli(env, **args):
     env.fout(output)
 
 
+def get_total_recurring_fee(args, result, table, total_hourly, total_monthly):
```
Comment (allmightyspiff) [inline SoftLayer/CLI/virt/create.py]: I don't like functions that modify the arguments that get passed in, good way to introduce bugs in my opinion. 

Lets change this function to `_build_reciept_table(prices)` which will build and return the whole table used to print costs. It doesn't need to know about hourly v monthly.

```python
def get_total_recurring_fee(prices, billing="hourly"):
    """Retrieve the total recurring fee of the items prices"""
    total = 0.0
    table = formatting.Table(['Cost', 'Item'])
    table.align['Cost'] = 'r'
    table.align['Item'] = 'l'
    for price in prices:
        rate = 0.0
        if billing == "hourly":
            rate += float(price.get('hourlyRecurringFee', 0.0))
        else:
            rate += float(price.get('recurringFee', 0.0))
        total += rate

        table.add_row(["%.3f" % rate, price['item']['description']])
    table.add_row(["%.3f" % total, "Total %s cost" % billing])
    return table
```

**prices need to be to 3 decimals**

File: SoftLayer/CLI/virt/create.py [hunk 2]
```diff
@@ -272,26 +272,27 @@ def cli(env, **args):
...
-        for price in result['prices']:
-            total_monthly += float(price.get('recurringFee', 0.0))
-            total_hourly += float(price.get('hourlyRecurringFee', 0.0))
-            if args.get('billing') == 'hourly':
-                rate = "%.2f" % float(price['hourlyRecurringFee'])
-            elif args.get('billing') == 'monthly':
-                rate = "%.2f" % float(price['recurringFee'])
+        if result['presetId']:
+            ordering_mgr = SoftLayer.OrderingManager(env.client)
+            preset_prices = ordering_mgr.get_preset_prices(result['presetId'])
```
Comment (allmightyspiff) [inline SoftLayer/CLI/virt/create.py]: If they are ordering a preset, what you should do is get the preset prices, and then add the server and ram items to the `results['prices']` array. It looks like the receipt already has the disk prices so no worries there.  

```python
if result['presetId']:
    ordering_mgr = SoftLayer.OrderingManager(env.client)
    preset_prices = ordering_mgr.get_preset_prices(result['presetId'])
    search_keys = ["guest_core", "ram"]
    for price in preset_prices['prices']:
        if price['item']['itemCategory']['categoryCode'] in search_keys:
            result['prices'].append(price)
```

Comment (allmightyspiff) [reply SoftLayer/CLI/virt/create.py]: Why the result data doesn't have the core/ram info is a mystery to me, but that was the best I could think of to get that data but NOT get all the other price data that gets returned.

**Comment by allmightyspiff** [comment]
While testing I made some changes that I think work, your welcome to apply them if it makes it easier
[patch.txt](https://github.com/softlayer/softlayer-python/files/2368672/patch.txt)

**Comment by allmightyspiff** [comment]
screen shot 2018-09-10 at 6 01 32 pm


The price for RAM+CPU from the API doens't seem to match up with the portal.
Portal shows price for BL2_4X16X300 though.... I'll have to see whats going on with that.

File: tests/managers/ordering_tests.py [hunk 3]
```diff
@@ -68,6 +68,57 @@ def test_get_package_id_by_type_returns_valid_id(self):
 
         self.assertEqual(46, package_id)
 
+    def test_get_preset_prices(self):
+        preset_id = 405
+        preset_prices = self.ordering.get_preset_prices(preset_id)
+
+        self.assertEqual(preset_prices, {
```
[the remark this answers is not in the document]
Comment (FernandoOjeda) [reply tests/managers/ordering_tests.py]: Done.

File: SoftLayer/CLI/virt/create.py [hunk 4]
```diff
@@ -272,26 +272,27 @@ def cli(env, **args):
         result = vsi.verify_create_instance(**data)
         total_monthly = 0.0
         total_hourly = 0.0
+        total_preset_monthly = 0.0
+        total_preset_hourly = 0.0
```
[the remark this answers is not in the document]
Comment (FernandoOjeda) [reply SoftLayer/CLI/virt/create.py]: Done.

File: SoftLayer/CLI/virt/create.py (hunk 1 above)
Comment (FernandoOjeda) [reply SoftLayer/CLI/virt/create.py]: Done.

File: SoftLayer/CLI/virt/create.py (hunk 2 above)
Comment (FernandoOjeda) [reply SoftLayer/CLI/virt/create.py]: Done.

**Comment by FernandoOjeda** [comment]
Now the total cost is right.
image

**Pull request merged**